File: AdminPanel1/AdminPanel1.Web/Modules/ServerTypes/SouqCom/CartForm.ts
import { StringEditor, ServiceLookupEditor, IntegerEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface CartForm {
    UserId: StringEditor;
    ProductId: ServiceLookupEditor;
    Qty: IntegerEditor;
}

export class CartForm extends PrefixedContext {
    static readonly formKey = 'SouqCom.Cart';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CartForm.init)  {
            CartForm.init = true;

            var w0 = StringEditor;
            var w1 = ServiceLookupEditor;
            var w2 = IntegerEditor;

            initFormType(CartForm, [
                'UserId', w0,
                'ProductId', w1,
                'Qty', w2
            ]);
        }
    }
}

File: AdminPanel1/AdminPanel1.Web/Modules/SouqCom/Cart/CartForm.cs
using Serenity.ComponentModel;

namespace AdminPanel1.SouqCom.Forms;

[FormScript("SouqCom.Cart")]
[BasedOnRow(typeof(CartRow), CheckNames = true)]
public class CartForm
{
    public string UserId { get; set; }
    public int ProductId { get; set; }
    public int Qty { get; set; }
}